core/src/main/java/hudson/model/ChoiceParameterDefinition.java

```diff
@@ -51,6 +56,58 @@ private ChoiceParameterDefinition(String name, List<String> choices, String defa
         this.defaultValue = defaultValue;
     }
 
+    /**
+     *
+     * @param name
+     * @param description
+     *
+     * @since TODO
+     */
+    @DataBoundConstructor
+    @Restricted(NoExternalUse.class) // there are specific constructors with String and List arguments for 'choices'
+    public ChoiceParameterDefinition(String name, String description) {
+        super(name, description);
+        this.choices = new ArrayList<>();
+        this.defaultValue = null;
+    }
+
+    /**
+     * Set the list of choices. Legal arguments are String (in which case the arguments gets split into lines) and Collection
+     * which sets the list of legal parameters to the String representations of the argument's non-null entries.
+     *
+     * See JENKINS-26143 for background.
+     *
+     * This retains the compatibility with the legacy String 'choices' parameter, while supporting the list type as generated
+     * by the snippet generator.
+     *
+     * @param choices String or Collection representing this parameter definition's possible values.
+     *
+     * @since TODO
+     *
+     */
+    @DataBoundSetter
+    @Restricted(NoExternalUse.class) // this is terrible enough without being used anywhere
+    public void setChoices(Object choices) {
+        if (choices instanceof String) {
+            setChoicesText((String) choices);
+            return;
+        }
+        if (choices instanceof Collection) {
+            this.choices = new ArrayList<>();
+            for (Object o : (Collection) choices) {
+                if (o != null) {
+                    this.choices.add(o.toString());
```

Comment:
Better to cast to `String` so people do not mistakenly pass a list of something else.

Reply:
TBH I would consider that a feature for use in Pipeline DSL, rather than a bug. Especially given [lack of star-dot in the sandbox](https://issues.jenkins-ci.org/browse/JENKINS-33051).